languages/emf-metamodel/src/main/java/de/jplag/emf/util/EmfaticModelView.java

```diff
@@ -0,0 +1,194 @@
+            return lastLineIndex;
+        }
+        lastLineIndex = line;
+        return line;
+    }
+
+    /**
+     * Locates the end index (closing character) for an element with a already known declaration index.
+     */
+    private int findEndIndexOf(int declarationIndex) {
+        int indentation = indentationOf(lines.get(declarationIndex));
+        if (declarationIndex > 1) { // exception for top level package
+            for (int i = declarationIndex + 1; i < lines.size(); i++) {
+                String nextLine = lines.get(i);
+                if (nextLine.length() > indentation && CLOSING_CHAR == nextLine.charAt(indentation)) {
+                    return i;
+                }
+            }
+        }
+        return lastLineIndex;
+    }
+
+    /**
+     * Checks if a token is representing a end of a block, e.g. a closing bracket.
+     */
+    private boolean isEndToken(Token token) {
+        return token.getType()instanceof MetamodelTokenType type && type.isEndToken();
+    }
+
+    /**
+     * Calculates the indentation of a line, meaning the number of tabs and spaces.
+     */
+    private int indentationOf(String beginLine) {
+        return beginLine.indexOf(beginLine.stripLeading());
+    }
+
+    /**
+     * Returns the line index of the declaration of an element in the Emfatic code.
+     */
+    private int lineIndexOf(ENamedElement element) {
+        return elementToLine.computeIfAbsent(element, this::findLineIndexOf);
+    }
+
+    /**
+     * Searches for the declaration of an element in the Emfatic code.
+     */
+    private int findLineIndexOf(ENamedElement element) {
+        String hash = Integer.toString(modelCopier.get(element).hashCode());
+        for (int index = 0; index < hashedLines.size(); index++) {
+            String line = hashedLines.get(index);
+            String trimmedLine = line.substring(indentationOf(line));
+            if (line.contains(hash) && isDeclaration(element, hash, trimmedLine)) {
+                return index;
+            }
+        }
+        return Token.NO_VALUE;
+    }
+
+    /**
+     * Checks if a line (with leading whitespace removed) contains an element based on the elements hash.
+     */
+    private boolean isDeclaration(ENamedElement element, String hash, String line) {
+        return isStructuralFeature(element, hash, line) || isEnumLiteral(element, hash, line) || isType(hash, line);
+    }
+
+    private boolean isType(String hash, String line) {
+        return line.matches(TYPE_KEYWORD_REGEX + hash + TYPE_SUFFIX_REGEX + ANYTHING_REGEX);
+    }
+
+    private boolean isEnumLiteral(ENamedElement element, String hash, String line) {
+        return element instanceof EEnumLiteral && line.matches(hash + ANYTHING_REGEX);
+    }
+
+    private boolean isStructuralFeature(ENamedElement element, String hash, String line) {
+        return element instanceof ETypedElement && line.matches(FEATURE_KEYWORD_REGEX + hash + ANYTHING_REGEX);
+    }
```

Comment:
Refactor this `+ ANYTHING_REGEX` into a method `boolean prefixMatches(String line, String regex)`